# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1364950120 14400
#      Tue Apr 02 20:48:40 2013 -0400
# Node ID 61989cde13aed9d3c644cbba3e2919400f9dd4dc
# Parent  5be43435bd5b919b44d14c6a3781be5c3be7e2c2
Use legacy implementation for strsplit().

* scripts/pkg/private/list_forge_packages.m: Specify delimitertype = "legacy"
* scripts/io/strread.m: ditto

diff --git a/scripts/io/strread.m b/scripts/io/strread.m
--- a/scripts/io/strread.m
+++ b/scripts/io/strread.m
@@ -767,17 +767,17 @@ function out = split_by (text, sep, mult
     ## However watch out if eol_char is also in delimiters
     if (index (sep, eol_char)); enchr = char (255); endif
     text = strrep (text, eol_char, [enchr eol_char enchr]);
   else
     mult_dlms_s1 = false;
   endif
 
   ## Split text string along delimiters
-  out = strsplit (text, num2cell (sep(:)), mult_dlms_s1);
+  out = strsplit (text, sep, mult_dlms_s1, "delimitertype", "legacy");
   if (index (sep, eol_char)); out = strrep (out, char (255), ''); endif
   ## In case of trailing delimiter, strip stray last empty word
   if (!isempty (out) && any (sep == text(end)))
     out(end) = [];
   endif
 
   ## Empty cells converted to empty cellstrings.
   out(cellfun ("isempty", out)) = {""};
diff --git a/scripts/pkg/private/list_forge_packages.m b/scripts/pkg/private/list_forge_packages.m
--- a/scripts/pkg/private/list_forge_packages.m
+++ b/scripts/pkg/private/list_forge_packages.m
@@ -20,17 +20,17 @@
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {@var{list} =} list_forge_packages ()
 ## Undocumented internal function.
 ## @end deftypefn
 
 function list = list_forge_packages ()
   [list, succ] = urlread ("http://packages.octave.org/list_packages.php");
   if (succ)
-    list = strsplit (list, " \n\t", true);
+    list = strsplit (list, " \n\t", true, "delimitertype", "legacy");
   else
     error ("pkg: could not read URL, please verify internet connection");
   endif
   if (nargout == 0)
     page_screen_output (false, "local");
     puts ("OctaveForge provides these packages:\n");
     for i = 1:length (list)
       try
